pkg/kv/kvserver/flow_control_integration_test.go

```diff
@@ -4917,6 +4917,84 @@ ORDER BY streams DESC;
 	h.query(n1, flowPerStoreTokenQueryStr, flowPerStoreTokenQueryHeaderStrs...)
 }
 
+func TestFlowControlRepeatedlySwitchMode(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
+	ctx := context.Background()
+	settings := cluster.MakeTestingClusterSettings()
+	tc := testcluster.StartTestCluster(t, 3, base.TestClusterArgs{
+		ReplicationMode: base.ReplicationManual,
+		ServerArgs: base.TestServerArgs{
+			Settings: settings,
+			RaftConfig: base.RaftConfig{
+				// Suppress timeout-based elections. This test doesn't want to
+				// deal with leadership changing hands.
+				RaftElectionTimeoutTicks: 1000000,
+			},
+			Knobs: base.TestingKnobs{
+				Store: &kvserver.StoreTestingKnobs{
+					FlowControlTestingKnobs: &kvflowcontrol.TestingKnobs{
+						UseOnlyForScratchRanges: true,
+						OverrideTokenDeduction: func(_ kvflowcontrol.Tokens) kvflowcontrol.Tokens {
+							// This test makes use of (small) increment requests, but
+							// wants to see large token deductions/returns.
+							return kvflowcontrol.Tokens(1 << 20 /* 1MiB */)
+						},
+					},
+				},
+			},
+		},
+	})
+	defer tc.Stopper().Stop(ctx)
+
+	// Setup the test state with 3 voters, one on each of the three
+	// node/stores.
+	k := tc.ScratchRange(t)
+	tc.AddVotersOrFatal(t, k, tc.Targets(1, 2)...)
+	h := newFlowControlTestHelperV2(t, tc, kvflowcontrol.V2EnabledWhenLeaderV2Encoding)
+	mode := kvflowcontrol.ApplyToElastic
+	h.init(mode)
+
+	desc, err := tc.LookupRange(k)
+	require.NoError(t, err)
+	h.waitForConnectedStreams(ctx, desc.RangeID, 3, 0 /* serverIdx */)
+	h.resetV2TokenMetrics(ctx)
+
+	finishedCh := make(chan struct{}, 1)
```

Comment:
nit: can `make(chan struct{})` and `defer close(finishedCh)` inside the goroutine, so that it's more apparent that it fires once.

Reply:
Done
<!-- Reviewable comment -OC64n35-M0OORMWmLJw:bb6un1j -->
<!-- Sent from Reviewable.io -->